manifests/server/replication.pp

```diff
@@ -0,0 +1,41 @@
+#
+class mysql::server::replication {
+  $master_slave  = $mysql::server::master_slave
+  $server_id     = $mysql::server::rep_server_id
+
+  if $mysql::server::replication_enable {
+    
+    exec { 'install-replication-plugin':
+      command => "mysql -u root -e \"INSTALL PLUGIN rpl_semi_sync_${mysql::server::master_slave} SONAME 'semisync_${mysql::server::master_slave}.so';\"",
```

Comment:
https://github.com/puppetlabs/puppetlabs-mysql/blob/master/manifests/params.pp has a variable for where the `my.cnf` file is located. Please use that instead of hardcoding a single location.